contracts/src/L2StandardBridgeBot.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity 0.8.20;
+
+import { Ownable } from "openzeppelin-contracts/contracts/access/Ownable.sol";
+
+interface IL2StandardBridge {
+    function withdrawTo(
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    ) external payable;
+}
+
+contract L2StandardBridgeBot is Ownable {
+    address public constant L2_STANDARD_BRIDGE_ADDRESS = 0x4200000000000000000000000000000000000010;
+    IL2StandardBridge public L2_STANDARD_BRIDGE = IL2StandardBridge(payable(L2_STANDARD_BRIDGE_ADDRESS));
+
+    address internal constant LEGACY_ERC20_ETH = 0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000;
+
+    uint256 public delegationFee;
+
+    event WithdrawTo(address indexed from, address l2Token, address to, uint256 amount, uint32 minGasLimit, bytes extraData);
+
+    receive() external payable { }
+
+    fallback() payable external { }
+
+    constructor(address payable _owner, uint256 _delegationFee) Ownable(_owner) {
+        delegationFee = _delegationFee;
+    }
+
+    // withdrawTo withdraws the _amount of _l2Token to _to address.
+    function withdrawTo(
+        address _l2Token,
+        address _to,
+        uint256 _amount,
+        uint32 _minGasLimit,
+        bytes calldata _extraData
+    ) public payable {
```

Comment:
The user may compose malicious tx for the bot runner to finalize, and these limitations can't be hardcoded in the contracts. So we should set filter for events to handle:
  - `_l2Token` in known list
  - `_minGasLimit` can't be bigger than known value. In practice, we'll set a fixed value in the front end
  - `_extraData` should be empty

txns from bridge UI will match the filter while custom txn sent directly by users may not.

Reply:
Hi @owen-reorg I have created another PR https://github.com/bnb-chain/opbnb-bridge-bot/pull/8

With https://github.com/bnb-chain/opbnb-bridge-bot/pull/8, I think it can handle malicious cases alraedy.

Then I think this suggesting filter may be not necessary for a malicious withdrawal, since the contract will return a revert error when proving or finalizing https://github.com/bnb-chain/opbnb-bridge-bot/pull/8/files#diff-279bc857f9ebbe8ad5b753f321c54f1e4ce671d6f489831783e71cbf46e3c69cR97. Afterward, we will record the revert-error and skip the reverted withdrawals later.